eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/CredentialInfo.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.core.implementation.util.ImplUtils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Locale;
+
+public final class CredentialInfo {
+
+    private static final String ENDPOINT = "Endpoint=";
+    private static final String SHARED_ACCESS_KEY_NAME = "SharedAccessKeyName=";
+    private static final String SHARED_ACCESS_KEY = "SharedAccessKey=";
+    private static final String ENTITY_PATH = "EntityPath=";
+
+    private URI endpoint;
+    private String sharedAccessKeyName;
+    private String sharedAccessKey;
+    private String eventHubName;
+
+    private CredentialInfo() { }
+
+    /**
+     * Create a CredentialInfo object from connection string.
+     *
+     * @param connectionString Connection String, which should at least include Endpoint, SharedAccessKeyName and SharedAccessKey
+     * @return CredentialInfo
+     * @throws IllegalArgumentException when 'connectionString' is {@code null} or empty, or cannot be translated into an
+     * {@link CredentialInfo}, or have invalid format.
+     */
+    public static CredentialInfo from(String connectionString) {
+
+        CredentialInfo credentialInfo = new CredentialInfo();
+        if (ImplUtils.isNullOrEmpty(connectionString)) {
+            throw new IllegalArgumentException("connection string is null or empty.");
+        }
+
+        String[] args = connectionString.split(";");
+
+        for (String arg : args) {
+            String segment = arg.trim();
+            String lowerCaseSegment = segment.toLowerCase(Locale.ENGLISH);
+            if (lowerCaseSegment.startsWith(ENDPOINT.toLowerCase(Locale.ENGLISH))) {
```

Comment:
I think a smarter way to do this comparison is to do an additional split here on "=", and then compare the key of that path rather than the current way we are doing it.
Having all these .toLowerCase() makes it unreadable where we could be using .equalsIgnoreCase(). If you need a reference, look at the .NET ConnectionString.

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/src/Core/ConnectionStringParser.cs